mat64/triangular.go

```diff
@@ -115,7 +115,18 @@ func (t *TriDense) Dims() (r, c int) {
 }
 
 func (t *TriDense) Triangle() (n int, upper bool) {
-	return t.mat.N, t.mat.Uplo == blas.Upper
+	return t.mat.N, t.isUpper()
```

Comment:
I assume we want the following to continue working:
````
var t TriDense
t.Cholesky(a, true)
````
In this case, there will be cases where t has an invalid uplo. I think we should say something like "For a zero-sized matrix Triangle returns false by convention, though the orientation is flexible"